man/mkinitcpio.8.txt

```diff
@@ -59,14 +64,39 @@ Options
 *-L, \--listhooks*::
 	List all available hooks.
 
+*-l, \--splash* 'filename'::
+	UEFI executables can show a bitmap file on boot.
+
+*-I, \--uefistub* 'filename'::
+	UEFI stub image used for UEFI executable generation. Only used if '--uefi'
+	is specified. Default: Attempts to look for a systemd-boot or gummiboot
+	stub loader.
+
+*-i, \--kernelimage* 'filename'::
+	Include a kernel image for the UEFI executable. This is only used if
+	'--uefi' is specified.  Default: one of
+	'/lib/modules/$KERNELVERSION/vmlinuz', '/boot/vmlinuz-$KERNELVERSION', or
```

Comment:
Can't this be derived from the preset, if not specified? If I'm creating an UEFI executable for -lts, for example, the preset already lists what is the kernel image and the path.

Reply:
It is derived though, but we can't be sure where the kernel image is in all cases which is why we also expose the ability to override whatever mkinitcpio tries to figure out. `--kernel` which you are probably thinking about is for the kernel version, not the file itself.